src/test/ui/inference/cannot-infer-async-enabled-impl-trait-bindings.stderr

```diff
@@ -8,14 +8,13 @@ LL | #![feature(impl_trait_in_bindings)]
    = note: see issue #63065 <https://github.com/rust-lang/rust/issues/63065> for more information
 
 error[E0282]: type annotations needed for `impl Future`
-  --> $DIR/cannot-infer-async-enabled-impl-trait-bindings.rs:13:20
+  --> $DIR/cannot-infer-async-enabled-impl-trait-bindings.rs:15:9
    |
 LL |     let fut = async {
-   |         --- consider giving `fut` the explicit type `impl Future`, with the type parameters specified
-LL |         make_unit()?;
-   |                    ^ cannot infer type of error for `?` operator
-   |
-   = note: `?` implicitly converts the error value into a type implementing `From<std::io::Error>`
+   |         --- consider giving `fut` the explicit type `impl Future`, where the type parameter `E` is specified
+...
+LL |         Ok(())
+   |         ^^ cannot infer type for type parameter `E` declared on the enum `Result`
```

Comment:
@estebank I got some of the places fixed up, but there are some others like this where I've regressed the diagnostics, losing some of the special `?` notes.  If you have any pointers to share on how to get them fixed up, I'd appreciate it.

Reply:
To be fair, the existing `?` diagnostic was already incorrect: this seems to imply that the `async` block behaves similarly to `try`, which it doesn't. We might have to proactively look for `?` expressions in `async` block bodies and give a specific error for them, or collect their spans and see if any E0282s happen in the same block as one of them, to treat them explicitly. Don't think these should block landing this, but needs to be addressed before these errors land on stable.